activesupport/lib/active_support/time_with_zone.rb

```diff
@@ -220,7 +220,11 @@ def strftime(format)
 
     # Use the time in UTC for comparisons.
     def <=>(other)
-      utc <=> other
+      if other.class == self.class
```

Comment:
`other.is_a?(TimeWithZone)` is probably better here to cover inherited classes
I would also put a note that this is for performance.